### ISSUE TITLE
Mixxx uses german language on macOS despite system language set to english

### ISSUE THREAD

**Issue by foss-** [mixxxdj/mixxx#11195]
### Bug Description

Mixxx 2.4-alpha-1398-g9c4e2f1d1a (main)
macOS 13.1

macOS set to english as primary language with german as secondary language. Region set to germany in case that plays a role, although it shouldn't.
Mixxx set to Language: System

**Currently**

Mixxx UI is german.

**Expected**

Mixxx UI should be english since it would be expected for Mixxx to follow system language.

The simple workaround obviously is to just set Mixxx language to english. Still found this to be unexpected behavior and thus reporting.

### Version

2.4 alpha

### OS

macOS 13.1

**Comment by daschuer** [comment]
There is probably a bug in Qt that is fixed in Qt 6.2 
https://github.com/qt/qtbase/commits/24e8a693cdf418a90559472addfd49637ff4e270/src/corelib/text/qlocale_mac.mm

Maybe this one: 
https://github.com/qt/qtbase/commit/d8158c6c93fc2a83f5acacc7baaf1a840951cf1e
https://bugreports.qt.io/browse/QTBUG-90971

**Comment by daschuer** [comment]
The Qt call and the native getMacLocalName() return the same: 
```
Debug [Main]: QLocale::system() = QLocale(English, Latin, Germany)
Debug [Main]: getMacLocaleName() = "en_DE"
```

**Comment by foss-** [comment]
Additional testing:

- open ~/Library/Containers/org.mixxx.mixxx/Data/Library/Application Support/Mixxx/mixxx.cfg`
- edit `Locale` and use `Locale en_DE`
- save and start mixxx

Mixxx launches with en UI.

**Comment by daschuer** [comment]
What does `echo $LANG` return? For me it is `de_DE.UTF-8`

**Comment by ronso0** [comment]
Fixed by #11218
